test/unit/constants.test.ts

```diff
@@ -1,54 +1,96 @@
 import * as fs from "fs"
-import { commit, getPackageJson, version } from "../../src/node/constants"
 import { tmpdir } from "../../test/utils/constants"
 import { loggerModule } from "../utils/helpers"
 
 // jest.mock is hoisted above the imports so we must use `require` here.
 jest.mock("@coder/logger", () => require("../utils/helpers").loggerModule)
 
 describe("constants", () => {
-  describe("getPackageJson", () => {
-    afterEach(() => {
-      jest.clearAllMocks()
+  beforeAll(() => {
+    jest.clearAllMocks()
+    jest.resetModules()
+  })
+  describe("with package.json defined", () => {
+    const { getPackageJson } = require("../../src/node/constants")
+    let mockPackageJson = {
+      name: "mock-code-server",
+      description: "Run VS Code on a remote server.",
+      repository: "https://github.com/cdr/code-server",
+      version: "1.0.0",
+      commit: "f6b2be2838f4afb217c2fd8f03eafedd8d55ef9b",
+    }
+    let version = ""
+    let commit = ""
+
+    beforeEach(() => {
+      jest.mock("../../package.json", () => mockPackageJson, { virtual: true })
+      commit = require("../../src/node/constants").commit
+      version = require("../../src/node/constants").version
     })
 
     afterAll(() => {
-      jest.restoreAllMocks()
+      jest.clearAllMocks()
+      jest.resetModules()
     })
 
-    it("should log a warning if package.json not found", () => {
-      const expectedErrorMessage = "Cannot find module './package.json' from 'src/node/constants.ts'"
-
-      getPackageJson("./package.json")
-
-      expect(loggerModule.logger.warn).toHaveBeenCalled()
-      expect(loggerModule.logger.warn).toHaveBeenCalledWith(expectedErrorMessage)
+    it("should provide the commit", () => {
+      // Source: https://gist.github.com/jhorsman/62eeea161a13b80e39f5249281e17c39#gistcomment-2896416
+      expect(commit).toBe("f6b2be2838f4afb217c2fd8f03eafedd8d55ef9b")
     })
 
-    it("should find the package.json", () => {
-      // the function calls require from src/node/constants
-      // so to get the root package.json we need to use ../../
-      const packageJson = getPackageJson("../../package.json")
-      expect(Object.keys(packageJson).length).toBeGreaterThan(0)
-      expect(packageJson.name).toBe("code-server")
-      expect(packageJson.description).toBe("Run VS Code on a remote server.")
-      expect(packageJson.repository).toBe("https://github.com/cdr/code-server")
-    })
-  })
-  describe("version", () => {
     it("should return the package.json version", () => {
       // Source: https://gist.github.com/jhorsman/62eeea161a13b80e39f5249281e17c39#gistcomment-2896416
       const validSemVar = new RegExp("^(0|[1-9]d*).(0|[1-9]d*).(0|[1-9]d*)")
       const isValidSemVar = validSemVar.test(version)
       expect(version).not.toBe(null)
       expect(isValidSemVar).toBe(true)
+      expect(version).toBe("1.0.0")
```

Comment:
```suggestion
      expect(version).toBe(mockPackageJson.version)
```

Reply:
There are a few places we could do this, I'm not sure if it's better than just writing out the string though.